fuzz/fuzz_targets/fuzz_skrifa_outline.rs

```diff
@@ -154,11 +129,99 @@ fn do_glyf_things(outline_request: OutlineRequest, data: &[u8]) -> Result<(), Bo
+            lcd_subpixel: None,
+            preserve_linear_metrics: true,
+        },
+        HintingMode::Smooth {
+            lcd_subpixel: None,
+            preserve_linear_metrics: false,
+        },
+        HintingMode::Smooth {
+            lcd_subpixel: Some(LcdLayout::Horizontal),
+            preserve_linear_metrics: true,
+        },
+        HintingMode::Smooth {
+            lcd_subpixel: Some(LcdLayout::Horizontal),
+            preserve_linear_metrics: false,
+        },
+        HintingMode::Smooth {
+            lcd_subpixel: Some(LcdLayout::Vertical),
+            preserve_linear_metrics: true,
+        },
+        HintingMode::Smooth {
+            lcd_subpixel: Some(LcdLayout::Vertical),
+            preserve_linear_metrics: false,
+        },
+    ]
+}
+
 fuzz_target!(|data: &[u8]| {
-    let mut unstructured = Unstructured::new(data);
-    let Ok(outline_request) = unstructured.arbitrary() else {
+    // data from corpus is likely to be a font. If we chope off the head to make an outline request
+    // it is likely data is no longer a font. So, take the cross product of likely values for various options
+    // If a lot of values are possible choose randomly with rng seeded from data to ensure reproducible results.
+    let Ok(font) = FontRef::new(data) else {
         return;
     };
-    let data = unstructured.take_rest();
-    let _ = do_glyf_things(outline_request, data);
+    let mut requests = vec![OutlineRequest::default()];
+    requests = requests
+        .into_iter()
+        .flat_map(|r| {
+            fuzz_sizes()
+                .into_iter()
+                .map(move |size| OutlineRequest { size, ..r.clone() })
+        })
+        .flat_map(|r| {
+            fuzz_locations(&font)
+                .into_iter()
+                .map(move |location| OutlineRequest {
+                    location,
+                    ..r.clone()
+                })
+        })
+        .flat_map(|r| {
+            [true, false].into_iter().map(move |hinted| OutlineRequest {
+                hinted,
+                ..r.clone()
+            })
+        })
+        .flat_map(|r| {
+            hinting_modes(r.hinted)
+                .into_iter()
+                .map(move |hinting_mode| OutlineRequest {
+                    hinting_mode,
+                    ..r.clone()
+                })
+        })
+        .flat_map(|r| {
+            [true, false]
+                .into_iter()
+                .map(move |with_memory| OutlineRequest {
+                    with_memory,
+                    ..r.clone()
+                })
+        })
+        .flat_map(|r| {
+            if r.with_memory {
+                vec![1024, 4096, 16384]
+            } else {
+                vec![0]
+            }
+            .into_iter()
+            .map(move |memory_size| OutlineRequest {
+                memory_size,
+                ..r.clone()
+            })
+        })
+        .collect();
+
+    for request in requests {
```

Comment:
I've lost track, but how many different requests are we testing? I don't suppose we're worried about long runtime?

Reply:
I believe 96 * the location cross. I suppose when the fuzzer develops very large fvar tables that might be a problem lol.